### PR TITLE
[CBC] Fix title extraction (fixes issue #16502)

### PR DESCRIPTION
**Pull request by parmjitv**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

Updated the extractor for the cbc.ca website to properly fetch the video title - issue #16502.

From the page source, the order of extraction should be: 1) OpenGraph title if available, 2) Twitter title if available, 3) Document title otherwise.

Cheers,

Parmjit V.

### CODE REVIEW

File: youtube_dl/extractor/cbc.py [hunk 1]
```diff
@@ -129,15 +129,17 @@ def _extract_player_init(self, player_init, display_id):
     def _real_extract(self, url):
         display_id = self._match_id(url)
         webpage = self._download_webpage(url, display_id)
+        title = self._og_search_title(webpage, default=None) or self._html_search_meta(
+            'twitter:title', webpage, 'title', default=None) or self._html_search_regex(
+                r'<title>(.*?)</title>', webpage, 'title', fatal=True)
```
Comment (dstftw) [inline youtube_dl/extractor/cbc.py]: Playlist title must not be fatal. Do not capture empty strings.

**Comment by parmjitv** [comment]
@dstftw - Thanks for the prompt review.  The regex for the document title is now set as non-fatal.

I would expect the document title will be set in the page source to some non-trivial string, so the fatal flag here is somewhat superfluous.  Does this address your concern about capturing an empty string?

**Comment by dstftw** [comment]
Nothing changed, how is this even related to capturing empty strings?

**Comment by parmjitv** [comment]
@dstftw - Ah, of course my regex was off for empty strings.  Thanks for fixing that!

**Pull request merged**